src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -139,35 +140,54 @@ internal PEReader PEReaderOpt
             }
         }
 
-        internal unsafe MetadataReader MetadataReader
+        internal MetadataReader MetadataReader
         {
             get
             {
                 if (_lazyMetadataReader == null)
                 {
-                    MetadataReader newReader;
+                    InitializeMetadataReader();
+                }
+                else if (_isDisposed)
```

Comment:
Why not just make this an unconditional check?  In the current form it doesn't appear this will protect against this pattern:

``` 
var m = new MetadataReader();
m.Dispose();
m.MetadataReader ... 
```

Yeah that's a bit pathological.